TLM/TLM/Util/DetailLogger.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Text;
+using CSUtil.Commons;
+
+namespace TrafficManager.Util {
+    internal static class DetailLogger {
+
+        public delegate void LogDelegate(params object[] lines);
+
+        [Conditional("DEBUG")]
+        public static void LogDebug(string heading, params object[] lines) {
+            var entry = new StringBuilder();
+
+            if (heading != null) {
+                entry.Append(heading);
+                entry.Append(' ');
+            }
+            string newline = string.Empty;
+
+            foreach (var o in lines) {
+                entry.Append(newline);
+                newline = "\n";
+                if (o is string) {
+                    entry.Append(o);
+                } else {
+                    BuildBLock(o);
+                }
+            }
+            Log._Debug(entry.ToString());
+
+            void BuildBLock(object o) {
+                string separator = string.Empty;
+                foreach (var p in o.GetType().GetProperties()) {
+                    entry.Append(separator);
+                    separator = " ";
+                    if (p.Name[0] != '_' || p.Name.Any(c => c != '_')) {
```

Comment:
The first condition seems redundant.
```suggestion
                    if (p.Name.Any(c => c != '_')) {
```
This lambda is allocating heap per variable but that already happens with input arguments so it is fine.
maybe add a attribute to hide the warning?

Reply:
btw, what heap is being allocated by the lambda?

Reply:
I think making this GC friendly is a waste of time.

Reply:
And now that I'm thinking about it, the tricks won't work anyway because we're using a delegate. 🤷‍♂️

Reply:
You guys seem to be having fun making it GC friendly. But its mostly debug-only logging, and its controlled with a switch, so mind that :)

Reply:
> You guys seem to be having fun making it GC friendly.

I'm really not. 😆 I just did this one optimization because it was convenient and obvious. In general, I'll avoid a heap allocation if there is an obvious way to do so, but that wasn't my main priority here. 😉